### ISSUE TITLE
[no-shadow] false positive for enum member names

### ISSUE THREAD

**Issue by davidgilbertson** [typescript-eslint/typescript-eslint#2886]
This is similar to https://github.com/typescript-eslint/typescript-eslint/issues/2721.

That was closed because the _value_ of the enum member should legitimately trigger the lint error. This issue is specifically about the _name_ of an enum member.

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-shadow": "error",
  }
}
```

Minimal case:
```TS
const Foo = 'Bar';

enum MyEnum {
  Foo = 'Baz', // Error: 'Foo' is already declared in the upper scope.
}
```

The slightly more real world case is where I have an enum to define the type of thing, and commonly that's the same as a the thing in question.
```ts
import {Cat, Dog} from './outside';

enum AnimalType {
  Cat = 'Cat', // Error
  Dog = 'Dog', // Error
}

// Of course, the const equivalent is fine:
const ANIMAL_TYPE = {
  Cat: 'Cat',
  Dog: 'Dog',
};
```

**Expected Result**

No error

**Actual Result**

Error

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.4.1` |
| `@typescript-eslint/parser`        | `4.7.0` |
| `TypeScript`                       | `4.1.3` |
| `ESLint`                           | `7.14.0` |
| `node`                             | `14.14.0` |

**Comment by bradzacher** [comment]
You've misunderstood #2721.
It's about the enum members.

My comment explains exactly why this errors:
https://github.com/typescript-eslint/typescript-eslint/issues/2721#issuecomment-718804636

**Comment by davidgilbertson** [comment]
Oh, you're quite right, I read it a fifth time and see now what you mean about enums creating their own scope.

Sorry for the issue clutter!
